small-talk/components/smolAuth.js

```diff
@@ -0,0 +1,55 @@
+import { getCookie } from "@/util/smolCookie";
+
+export function smolAuth(getServerSidePropsFunc) {
+    return async (ctx) => {
+        const { req } = ctx;
+        
+        if (req.headers.cookie) {
+            const session_id = getCookie("session_id", req.headers.cookie);
+            //console.log(session_id)
+            try {
+                let response = await fetch("http://" + req.headers.host + "/api/auth2/", {
+                    method: "POST",
+                    headers: {
+                        "Content-Type" : "application/json",
+                        "Authorization" : session_id
+                    }
+                })
```

Comment:
Does using an API ensure cookies are created and held on the server side?

Reply:
This API route ensures the session ID is valid. A separate api route should be created to create/update a session id in the database